devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,599 @@
+	childDevice?.sendEvent(hueState)
+	childDevice?.sendEvent(name:"saturation", value:"100")
+}
+
+def createChildDevice(childDthNamespace, childDthName, childDni, childComponentLabel, childComponentName) {
+	try {
+		log.debug "Creating a child device: ${childDthNamespace}, ${childDthName}, ${childDni}, ${childComponentLabel}, ${childComponentName}"
+		def childDevice = addChildDevice(childDthNamespace, childDthName, childDni, device.hub.id,
+			[ completedSetup: true,
+			  label: childComponentLabel,
+			  isComponent: true,
+			  componentName : childComponentName,
+			  componentLabel: childComponentLabel
+			])
+		childDevice
+	} catch(Exception e) {
+		log.debug "Exception: ${e}"
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
+	dimmerEvents(cmd)
+}
+
+def dimmerEvents(physicalgraph.zwave.Command cmd) {
+	log.debug "dimmer events ${cmd}"
+	def map = [name : "switch", value : cmd.value ? "on" : "off", descriptionText : "$device.displayName was turned $cmd.value"]
+	def switchEvent = createEvent(map)
+	def result = [switchEvent]
+	if (cmd.value) {
+		map = [name: "level", value: cmd.value == 99 ? 100 : cmd.value , unit: "%"]
+		result << createEvent(map)
+	}
+	if (switchEvent.isStateChange) {
+		result << response(["delay 3000", zwave.meterV3.meterGet(scale: 2).format()])
+	}
+	return result
+}
+
+def on() {
+	encapSequence([
+		zwave.basicV1.basicSet(value: 0xFF),
+		zwave.basicV1.basicGet()
+	], 5000)
+}
+
+def off() {
+	encapSequence([
+		zwave.basicV1.basicSet(value: 0x00),
+		zwave.basicV1.basicGet()
+	], 5000)
+}
+
+def setLevel(level){
+	log.debug"setLevel"
+	if(level > 99) level = 99
+	encapSequence([
+		zwave.basicV1.basicSet(value: level),
+		zwave.switchMultilevelV1.switchMultilevelGet()
+	], 5000)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def map = [:]
+	if (cmd.meterType == 1 && cmd.scale == 0){
+		map = [name: "energy", value: cmd.scaledMeterValue, unit: "kWh"]
+	} else if(cmd.meterType == 1 && cmd.scale == 2){
+		map = [name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W"]
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	// Handles all Z-Wave commands we aren't interested in
+	log.debug "${cmd}"
+	[:]
+}
+
+def childSetColor(deviceNetworkId, value) {
+	def id = deviceNetworkId?.split(":")[1] as Integer
+	sendHubCommand setColorCmd(value, id)
+}
+
+def setColorCmd(value, id) {
+	if (value.hue == null || value.saturation == null) return
+	def ledColor = Math.round(huePercentToZwaveValue(value.hue))
+	encapSequence([
+		zwave.configurationV2.configurationSet(scaledConfigurationValue: ledColor, parameterNumber: 13, size: 2),
+		zwave.configurationV2.configurationGet(parameterNumber: 13)
+	], 5000)
+}
+
+private huePercentToZwaveValue(value){
+	return value<=2?0:(value>=98?255:value/100*255)
```

Comment:
put some spaces across those statements and conditions, so they'd be easier to read

Reply:
Done